pkg/router/controller/unique_host.go

```diff
@@ -119,11 +141,14 @@ func (p *UniqueHost) HandleRoute(eventType watch.EventType, route *routeapi.Rout
 		} else {
 			if oldest.CreationTimestamp.Before(route.CreationTimestamp) {
 				glog.V(4).Infof("Route %s cannot take %s from %s", routeName, host, routeNameKey(oldest))
-				return fmt.Errorf("route %s holds %s and is older than %s", routeNameKey(oldest), host, key)
+				err := fmt.Errorf("another route holds %s and is older than %s", host, route.Name)
```

Comment:
I changed the message slightly in
https://github.com/openshift/origin/pull/7388

On Wed, Feb 17, 2016 at 11:21 AM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> It's a security issue to show the other namespace name, but we can tell
> the user it's in a different namespace.
>
> On Wed, Feb 17, 2016 at 11:08 AM, Maru Newby <notifications@github.com>
> wrote:
>
>> In pkg/router/controller/unique_host.go
>> <https://github.com/openshift/origin/pull/6953#discussion_r53186546>:
>>
>> > @@ -119,11 +141,14 @@ func (p *UniqueHost) HandleRoute(eventType watch.EventType, route *routeapi.Rout
>> >  		} else {
>> >  			if oldest.CreationTimestamp.Before(route.CreationTimestamp) {
>> >  				glog.V(4).Infof("Route %s cannot take %s from %s", routeName, host, routeNameKey(oldest))
>> > -				return fmt.Errorf("route %s holds %s and is older than %s", routeNameKey(oldest), host, key)
>> > +				err := fmt.Errorf("another route holds %s and is older than %s", host, route.Name)
>>
>> Would it make sense for this error message to be explicit that the other
>> route is in a different namespace? It's clear to a programmer who can see
>> where the error is thrown, but a user might not have that advantage.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openshift/origin/pull/6953/files#r53186546>.
>>
>
>